src/components/Button.tsx

```diff
@@ -0,0 +1,18 @@
+import type { ButtonProps } from "./types";
+
+export const Button: React.FC<ButtonProps> = ({
```

Comment:
When you're building a standard, reusable Button component (especially in React + TypeScript), it's not just about onClick, type, etc. To make it truly flexible and scalable, your button should account for both built-in props (via HTMLButtonElement) and custom props to cover styling, behavior, accessibility, and consistency across your app.


```typescript
export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: 'primary' | 'secondary' | 'ghost' | 'danger';
  size?: 'sm' | 'md' | 'lg';
  icon?: React.ReactNode;
  iconPosition?: 'left' | 'right';
  isLoading?: boolean;
  loadingText?: string;
  fullWidth?: boolean;
  as?: 'button' | 'a';
  href?: string;
  rounded?: boolean | 'full' | 'md';
  tooltip?: string;
}
```